steward/src/grpc_client.rs

```diff
@@ -0,0 +1,158 @@
+use crate::error::{Error, ErrorKind};
+use futures::future::join_all;
+use somm_proto::cork::{query_client::QueryClient as QClient, QueryCommitPeriodRequest};
+use std::{thread, time::Duration};
+use steward_proto::{
+    self,
+    steward::{
+        contract_call_client::ContractCallClient as ContractClient, SubmitRequest, SubmitResponse,
+    },
+};
+use tonic::transport::{Channel, ClientTlsConfig, Uri};
+
+pub type ContractCallClient = ContractClient<Channel>;
+pub type QueryClient = QClient<Channel>;
+
+pub struct GrpcClient {
+    pub grpc_address: String,
+    pub steward_endpoints: Vec<String>,
+    pub tls_configuration: ClientTlsConfig,
+}
+
+impl GrpcClient {
+    pub fn check_for_grpc_address(&self) -> Result<(), Error> {
+        if self.grpc_address.is_empty() {
+            return Err(ErrorKind::GrpcError
+                .context("no grpc address available")
+                .into());
+        }
+
+        Ok(())
+    }
+
+    pub fn check_for_steward_addresses(&self) -> Result<(), Error> {
+        if self.steward_endpoints.is_empty() {
+            return Err(ErrorKind::GrpcError
+                .context("no grpc address available")
```

Comment:
I think this should be `No endpoint found` or `steward address unavailable`. Since we already have the `no grpc address available` message, I think duplicating it is a bad idea.

Reply:
Agreed, that'd make it clearer, will update